src/syscheckd/db/fim_db_files.c

```diff
@@ -535,45 +492,10 @@ void fim_db_remove_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex
         }
     }
 
-    w_mutex_unlock(mutex);
-
-
-    if (send_alert && rows >= 1) {
-        whodata_evt *whodata_event = (whodata_evt *) w_evt;
-        cJSON * json_event      = NULL;
-        char * json_formatted    = NULL;
-        int pos = 0;
-        const char *FIM_ENTRY_TYPE[] = {"file", "registry"};
-
-        // Obtaining the position of the directory, in @syscheck.dir, where @entry belongs
-        if (pos = fim_configuration_directory(entry->file_entry.path,
-            FIM_ENTRY_TYPE[entry->type]), pos < 0) {
-            goto end;
-        }
-
-        w_mutex_lock(mutex);
-        json_event = fim_json_event(entry->file_entry.path, NULL, entry->file_entry.data, pos, FIM_DELETE, mode,
-                                    whodata_event, NULL);
-        w_mutex_unlock(mutex);
-
-        if (!strcmp(FIM_ENTRY_TYPE[entry->type], "file") && syscheck.opts[pos] & CHECK_SEECHANGES) {
-            fim_diff_process_delete_file(entry->file_entry.path);
-        }
-
-        if (json_event) {
-            mdebug2(FIM_FILE_MSG_DELETE, entry->file_entry.path);
-            json_formatted = cJSON_PrintUnformatted(json_event);
-            send_syscheck_msg(json_formatted);
-
-            os_free(json_formatted);
-            cJSON_Delete(json_event);
-        }
-    }
 
 end:
-    w_mutex_lock(mutex);
     fim_db_check_transaction(fim_sql);
-    w_mutex_unlock(mutex);
+    return rows;
```

Comment:
Returning `rows` here does not make much sense, we should change the function to return `FIMDB_ERR` if any operation fails and `FIMDB_OK` if everything works as planned.